pkg/system/phase2_creating.go

```diff
@@ -130,6 +131,12 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 			return err
 		}
 	}
+
+	// Once the db is reconciled, upgrade it if it is required
+	if err := r.UpgradeDb(); err != nil {
```

Comment:
Upgrade DB should be performed before `ReconcileDB`

Reply:
Made changes as suggested